### PR TITLE
Accounting Ratios widgets

### PR DESCRIPTION
**Pull request by cesar-tonnoir**
https://github.com/maestrano/impac/pull/420

This needs to go to prod this week as part of the new widgets/kpis for demos

thx @xaun :beers:

### CODE REVIEW

File: src/components/widgets/accounts-ratios/accounts-ratios.tmpl.html [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        <div class="history chart-container" ng-show="widget.isHistoryMode">
+          <div impac-chart draw-trigger="::drawTrigger.promise" deferred="::chartDeferred"></div>
+          <div class="legend">{{ widget.content.layout.ratio }}</div>
+        </div>
+      </div>
+
+    </div>
+
+    <!-- No data found -->
+    <div ng-show="(isDataFound==false)" common-data-not-found on-display-alerts="onDisplayAlerts()" widget-engine="::widget.category" />
```
Comment (xaun) [inline src/components/widgets/accounts-ratios/accounts-ratios.tmpl.html]: This needs to be updated for the new scoped vars in the common-data-not-found directive.. I thought my search and replace updated the widget generator template :( 

It should be:
```
<div ng-show="(isDataFound==false)" common-data-not-found on-display-alerts="onDisplayAlerts()" endpoint="::widget.category" width="::widget.width" />
```

Comment (cesar-tonnoir) [inline src/components/widgets/accounts-ratios/accounts-ratios.tmpl.html]: yep, good pick-up. updated.

File: src/components/widgets/accounts-ratios/accounts-ratios.directive.coffee [hunk 2]
```diff
@@ -0,0 +1,98 @@
+#
+# Component generated by Impac! Widget Generator!
```
Comment (cesar-tonnoir) [inline src/components/widgets/accounts-ratios/accounts-ratios.directive.coffee]: ;)

**Comment by cesar-tonnoir** [comment]
@xaun thanks for the review. After having done a first round of testing, I've pushed a few more commits.
Could you please review those and merge the PR if all good for you?
I'm going to deploy the feature branch on UAT to keep the testing going in the meantime

File: src/components/widgets-settings/offsets/offsets.tmpl.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  <div ng-repeat="offsetValue in offsets track by $index" class="offset-value text-right">
+    <i class="fa fa-trash-o pull-left" ng-click="removeOffset($index)" title="Remove offset"/>
+    + <span ng-if="showIntervalsMult">{{ intervalsCount }} {{ periodWord }}{{ intervalsCount > 1 ? 's' : '' }} x </span>{{ offsetValue | mnoCurrency : currency}}
+  </div>
+
+  <div class="form-group">
+    <button class="btn btn-sm btn-success" ng-click="addOffset()" title="Add offset">
+      <i class="fa fa-plus" />
+    </button>
+    <input type="text" class="form-control input-sm pull-right text-right" placeholder="{{ period.charAt(0) + period.toLowerCase().slice(1) }} adjustment" ng-model="offsetFormula" ng-keyup="addOffsetOnEnter($event)">
```
Comment (xaun) [inline src/components/widgets-settings/offsets/offsets.tmpl.html]: The placeholder expression should probably be in a scoped method?

File: src/components/widgets/accounts-ratios/accounts-ratios.directive.coffee [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+  $scope.totalRatio = 0
+  $scope.totalNumerator = 0
+  $scope.totalDenominator = 0
+  $scope.calculatedNumerator = 0
+  $scope.calculatedDenominator = 0
+  $scope.simulationMode = false
+  $scope.intervalsCount = 0
+  $scope.isPnl = false
+  $scope.forwardParams = { histParams: w.metadata.hist_parameters }
```
Comment (xaun) [inline src/components/widgets/accounts-ratios/accounts-ratios.directive.coffee]: What is forwardParams ? I can see it contains hist parameters, but I don't get the naming (maybe a comment?)

File: src/components/kpi/kpi.directive.coffee [hunk 5]
```diff
@@ -176,6 +176,15 @@ angular
...
           ImpacKpisSvc.getKpiTargetPlaceholder($scope.kpi.endpoint, watchable)
 
+        $scope.getRealValue = ->
+          kpi = $scope.kpi
+          return "" if _.isEmpty(kpi.data)
+          value = kpi.data[kpi.watchables[0]].value
+          unit = kpi.data[kpi.watchables[0]].unit
+          valueArray = [value]
+          if (unit? && unit.length > 0) then valueArray.push(unit)
+          valueArray.join(' ')
```
Comment (cesar-tonnoir) [inline src/components/kpi/kpi.directive.coffee]: yep, makes sense. will do

File: src/components/widgets-settings/offsets/offsets.tmpl.html (hunk 3 above)
Comment (cesar-tonnoir) [inline src/components/widgets-settings/offsets/offsets.tmpl.html]: :+1:

File: src/components/widgets/accounts-ratios/accounts-ratios.directive.coffee (hunk 4 above)
Comment (cesar-tonnoir) [inline src/components/widgets/accounts-ratios/accounts-ratios.directive.coffee]: just copy-pasted what was done in another directive. I'll find something better

**Pull request merged**